docarray/array/storage/elastic/getsetdel.py

```diff
@@ -0,0 +1,85 @@
+import copy
+from typing import Iterable, Dict
+
+from ..base.getsetdel import BaseGetSetDelMixin
+from ..base.helper import Offset2ID
+from .... import Document
+
+
+class GetSetDelMixin(BaseGetSetDelMixin):
+    """Provide concrete implementation for ``__getitem__``, ``__setitem__``,
+    and ``__delitem__`` for ``DocumentArrayElastic``"""
+
+    def _getitem(self, doc_id: str) -> 'Document':
+        """Helper method for getting item with elastic as storage
+
+        :param doc_id:  id of the document
+        :raises KeyError: raise error when elastic id does not exist in storage
+        :return: Document
+        """
+        try:
+            result = self._client.get(index=self._config.index_name, id=doc_id)
+            doc = Document.from_base64(result['_source']['blob'])
+            doc.embedding = result['_source']['embedding']
+            return doc
+        except Exception as ex:
+            raise KeyError(doc_id) from ex
+
+    def _get_doc_by_id(self, _id: str) -> 'Document':
+        """Concrete implementation of base class' ``_get_doc_by_id``
+
+        :param _id: the id of the document
+        :return: the retrieved document from elastic
+        """
+        return self._getitem(_id)
+
+    def _set_doc_by_id(self, _id: str, value: 'Document'):
+        """Concrete implementation of base class' ``_set_doc_by_id``
+
+        :param _id: the id of doc to update
+        :param value: the document to update to
+        """
+        doc_copy = copy.deepcopy(value)
+        embedding = value.embedding
+        doc_copy.embedding = None
```

Comment:
we don' do this in other storage backends as we have to keep a version of embeddings within the serialized documents.
This is because the indexed embeddings can be different from the actual ones (e.g adding noise).
what would happen if we call _get_doc_by_id(id).embedding right after _set_doc_by_id ?